Signal/src/util/ThreadUtil.m

```diff
@@ -361,7 +369,15 @@ + (ThreadDynamicInteractions *)ensureDynamicInteractionsForThread:(TSThread *)th
             shouldHaveBlockOffer = NO;
         }
 
+        if (![OWSProfileManager.sharedManager hasLocalProfile] ||
+            [OWSProfileManager.sharedManager isThreadInProfileWhitelist:thread]) {
+            // Don't show offer if thread is local user hasn't configured their profile.
+            // Don't show offer if thread is already in profile whitelist.
+            shouldHaveAddToProfileWhitelistOffer = NO;
+        }
+
         // We use these offset to control the ordering of the offers and indicators.
+        const int kAddToProfileWhitelistOfferOffset = -4;
```

Comment:
There's a few open questions here.  One is: if "non-contact" Alice starts a new conversation with Bob, should Bob see three offers (block, add to contacts, share profile)?  That's a lot.  But I could see a user doing any one of those three and not the others.